core/state/state_test.go

```diff
@@ -567,3 +613,21 @@ func interacted(c interactionCounter, times int) func() bool {
 		return c.interactions() == times
 	}
 }
+
+type mockBalanceProvider struct {
+	Balance uint64
+}
+
+// GetBalance returns a pre-defined balance.
+func (mbp *mockBalanceProvider) GetBalance(_ identity.Identity) uint64 {
```

Comment:
This is identical to `mocks.BalanceProvider`

Reply:
I moved mocks next to tests, because keeping them global packages causes circular dependencies: `pingpong -> mocks -> connections -> pingpong`

Reply:
There is no dependency on `connections` in the mocks package. Either way, mocks should have minimal dependencies.

Reply:
I added similar mock `EarningsProvider.SettlementState(_ identity.Identity) pingpong.SettlementState` and `mocks` package starts to be interdependent.
Lets agree on having mocks next to package we're testing.

Reply:
Such widely used types as `identity.Identity` or in this case `pingpong.SettlementState` could be in a leaf package with minimal dependencies - that will solve the package cycle. If it's a one-off mock - then it's good to have it next to the test. If it's used multiple times, then the maintenance becomes a burden and it should be in `mocks`.